--- cups-filters-1.0.41/utils/cups-browsed.c.orig	2013-10-29 10:53:53.000000000 +0100
+++ cups-filters-1.0.41/utils/cups-browsed.c	2013-12-01 08:39:32.055087675 +0100
@@ -227,16 +227,6 @@
 }
 
 int
-ippSetVersion(ipp_t *ipp, int major, int minor)
-{
-  if (!ipp || major < 0 || minor < 0)
-    return (0);
-  ipp->request.any.version[0] = major;
-  ipp->request.any.version[1] = minor;
-  return (1);
-}
-
-int
 ippSetVersion(ipp_t *ipp,
               int   major,
               int   minor)
