From 2561a9dc5ebdfd90cc2781ca8f1560f4faff19a1 Mon Sep 17 00:00:00 2001
From: Till Kamppeter <till.kamppeter@gmail.com>
Date: Tue, 8 May 2018 20:57:56 +0200
Subject: [PATCH] pdftoopvp: Silence compiler warning.

---
 NEWS                               | 1 +
 filter/pdftoopvp/OPVPOutputDev.cxx | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index e29e3ba8..19caf601 100644
--- a/NEWS
+++ b/NEWS
@@ -3,6 +3,7 @@ NEWS - OpenPrinting CUPS Filters v1.20.3 - 2018-04-12
 
 CHANGES IN V1.20.4
 
+	- pdftoopvp: Silence compiler warning (Issue #42).
 	- cups-browsed: If the user modifies/overwrites a print queue
 	  created by cups-browsed, it will now not only be
 	  automatically released from the control of cups-browsed, but
diff --git a/filter/pdftoopvp/OPVPOutputDev.cxx b/filter/pdftoopvp/OPVPOutputDev.cxx
index 63806e8a..45ebe02c 100644
--- a/filter/pdftoopvp/OPVPOutputDev.cxx
+++ b/filter/pdftoopvp/OPVPOutputDev.cxx
@@ -66,7 +66,7 @@ struct SplashOutFontSubst {
 class SplashOutFontFileID: public SplashFontFileID {
 public:
 
-  SplashOutFontFileID(Ref *rA) { r = *rA; substIdx = -1; }
+  SplashOutFontFileID(const Ref *rA) { r = *rA; substIdx = -1; }
 
   ~SplashOutFontFileID() {}
 
